From fd9feb5c5c6e6b3e4a5a5c7bd663e02cf408a89b Mon Sep 17 00:00:00 2001
From: Maarten Lankhorst <maarten.lankhorst@canonical.com>
Date: Fri, 11 Oct 2013 20:32:08 +0200
Subject: [PATCH 02/40] avrt: Add realtime to stub

---
 dlls/avrt/main.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/dlls/avrt/main.c b/dlls/avrt/main.c
index 5c9ef4a..43e8bd8 100644
--- a/dlls/avrt/main.c
+++ b/dlls/avrt/main.c
@@ -80,6 +80,7 @@ HANDLE WINAPI AvSetMmThreadCharacteristicsW(LPCWSTR TaskName, LPDWORD TaskIndex)
         SetLastError(ERROR_INVALID_HANDLE);
         return NULL;
     }
+    SetThreadPriority(GetCurrentThread(), THREAD_PRIORITY_TIME_CRITICAL);
     return (HANDLE)0x12345678;
 }
 
-- 
1.8.3.2

