diff -rupN sumwars-code/packaging/sumwars.desktop sumwars-code.new/packaging/sumwars.desktop
--- sumwars-code/packaging/sumwars.desktop	2012-02-28 20:37:40.480886930 -0600
+++ sumwars-code.new/packaging/sumwars.desktop	2012-03-03 12:36:15.311116712 -0600
@@ -1,11 +1,30 @@
 [Desktop Entry]
 Version=1.0
+Encoding=UTF-8
 Type=Application
-Name=SumWars
-GenericName=Summoning Wars
+Name=Summoning Wars
+Name[de]=Zusammenrufen Kriegen
+Name[it]=Guerre Summoning
+Name[pl]=Wars Summoning
+Name[pt]=Guerras de Evocação
+Name[ru]=Вызов войны
+Name[uk]=Виклик війни
+GenericName=Role-Playing Game
+GenericName[de]=Role-Playingspiel
+GenericName[it]=Gioco di ruolo
+GenericName[pl]=Gry RPG
+GenericName[pt]=Jogo de Role-Playing
+GenericName[ru]=Ролевая игра
+GenericName[uk]=Рольова гра
 Comment=A multi-player, 3D Action Role-Playing Game
-TryExec=/usr/games/sumwars
-Exec=/usr/games/sumwars
-Categories=Game;RolePlaying;
-Icon=/usr/share/icons/hicolor/128x128/sumwars.png
+Comment[de]=Eine Mehrfach-Player, dreidimensionale Action-Rollenspiel gam
+Comment[it]=Un giocatore multiple, tre ruolo d'azione tridimensionale gioco gam
+Comment[pl]=Wielu graczy, trójwymiarowy rola akcja gry Gam
+Comment[pt]=Um jogador múltiplo, três papel de ação dimensional jogando gam
+Comment[ru]=Несколько игроков, трехмерные роль играет действие гамма
+Comment[uk]=Кілька гравців, тривимірні роль відіграє дію гамма
+Exec=sumwars
+TryExec=sumwars
+Icon=sumwars
 Terminal=false
+Categories=Game;RolePlaying;
