diff -rupN a/source/darkplaces/makefile.inc b/source/darkplaces/makefile.inc
--- a/source/darkplaces/makefile.inc	2013-06-05 03:03:36.000000000 -0500
+++ b/source/darkplaces/makefile.inc	2013-07-04 00:27:40.944905309 -0500
@@ -22,7 +22,7 @@ CC?=gcc
 # Experimental
 #CPUOPTIMIZATIONS?=-fno-math-errno -ffinite-math-only -fno-rounding-math -fno-signaling-nans -fassociative-math -freciprocal-math -fno-signed-zeros -fno-trapping-math
 # Normal
-CPUOPTIMIZATIONS?=-fno-math-errno -ffinite-math-only -fno-rounding-math -fno-signaling-nans -fno-trapping-math
+#CPUOPTIMIZATIONS?=-fno-math-errno -ffinite-math-only -fno-rounding-math -fno-signaling-nans -fno-trapping-math
 # NOTE: *never* *ever* use the -ffast-math or -funsafe-math-optimizations flag
 
 # Additional stuff for libode
