diff -rupN libircclient-1.6/configure.in libircclient-1.6.new/configure.in
--- libircclient-1.6/configure.in	2012-01-21 18:46:50.000000000 -0600
+++ libircclient-1.6.new/configure.in	2012-02-11 19:11:30.491475461 -0600
@@ -61,9 +61,7 @@ AC_CHECK_FUNCS([localtime_r socket])
 AC_CHECK_LIB(socket, socket, AC_DEFINE(HAVE_SOCKET) LIBS="$LIBS -lsocket")
 
 if test "$use_debug" = "yes"; then
-	CFLAGS="${CFLAGS} -g -DENABLE_DEBUG"
-else
-	CFLAGS="${CFLAGS} -O3"
+	CFLAGS="${CFLAGS} -DENABLE_DEBUG"
 fi
 
 # Choosing the target
@@ -76,14 +74,14 @@ fi
 if test "$use_ipv6" = "yes"; then
 	AC_CHECK_FUNCS([getaddrinfo inet_pton])
 	AC_CHECK_LIB(nsl, getaddrinfo, AC_DEFINE(HAVE_GETADDRINFO) LIBS="$LIBS -lnsl")
-	CFLAGS="$CFLAGS -DENABLE_IPV6"
+	CFLAGS="${CFLAGS} -DENABLE_IPV6"
 else
 	AC_CHECK_FUNCS([gethostbyname_r inet_ntoa])
 	AC_CHECK_LIB(nsl, inet_ntoa, AC_DEFINE(HAVE_INET_NTOA) LIBS="$LIBS -lnsl")
 fi
 
 if test "$use_threads" = "yes"; then
-	CFLAGS="$CFLAGS -DENABLE_THREADS -D_REENTRANT"
+	CFLAGS="${CFLAGS} -DENABLE_THREADS -D_REENTRANT"
 fi
 
 
