--- configuration.xml	2011-09-16 16:22:28.588608300 -0500
+++ configuration.xml.linux	2011-09-16 16:39:56.149113349 -0500
@@ -1,9 +1,9 @@
 <?xml version="1.0" standalone="yes"?>
 <configuration>
 	<title value="MegaGlest"/>
-	<file-name-main value="glest.ini"/>
-	<file-name value="$APPDATA\megaglest\glestuser.ini"/>
-	<icon value="true" path="megaglest.ico"/>
+	<file-name-main value="$APPLICATIONDATAPATH/glest.ini"/>
+	<file-name value="$HOME/.megaglest/glestuser.ini"/>
+	<icon value="true" path="$APPLICATIONDATAPATH/glest.ico"/>
 	<field-groups>
 		<field-group name="General">
 			<field type="Int">
@@ -354,16 +354,6 @@
 				<description value=""/>
 				<default value="false"/>
 				<enums>
-					<enum value="true"/>
-					<enum value="false"/>
-				</enums>
-			</field>
-			<field type="Enum">
-				<name value="Client and Server data file transfer support"/>
-				<variable-name value="AllowDownloadDataSynch"/>
-				<description value=""/>
-				<default value="false"/>
-				<enums>
 					<enum value="true"/>
 					<enum value="false"/>
 				</enums>
